subsys/net/ip/l2/ethernet/arp.c

```diff
@@ -290,22 +289,9 @@ struct net_pkt *net_arp_prepare(struct net_pkt *pkt)
 		net_sprint_ll_addr(ll->addr, sizeof(struct net_eth_addr)),
 		net_sprint_ipv4_addr(&NET_IPV4_HDR(pkt)->src));
 
-	frag = pkt->frags;
-	while (frag) {
-		/* If there is no room for link layer header, then
-		 * just send the packet as is.
-		 */
-		if (!net_buf_headroom(frag)) {
-			frag = frag->frags;
-			continue;
-		}
-
-		hdr = net_eth_fill_header(ctx, pkt, frag,
-					  htons(NET_ETH_PTYPE_IP),
-					  ll->addr, entry->eth.addr);
-
-		frag = frag->frags;
-	}
+	hdr = net_eth_fill_header(ctx, pkt, pkt->frags,
```

Comment:
While you're cleaning this up, `hdr` doesn't seem to be used anywhere else in this function.

Reply:
some vlan changes left over indeed.